tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature

```diff
@@ -151,63 +151,66 @@ Feature: Share by public link
+    And user "user1" has logged in using the webUI
+    When the user creates a new public link for folder "simple-folder" using the webUI with
+      | role | Editor |
     Then the user should see an error message on the public link share dialog saying "Passwords are enforced for link shares"
     And the public link should not have been generated
 
-  @skip @yetToImplement
-  Scenario: user tries to create a public link with write only permission without entering share password while enforce password on write only public share is enforced
-    Given parameter "shareapi_enforce_links_password_write_only" of app "core" has been set to "yes"
-    When the user tries to create a new public link for folder "simple-folder" using the webUI with
-      | permission | upload |
+  Scenario: user tries to create a public link with write only permission without entering share password while enforce password on upload-only(File Drop) is enforced
+    Given the setting "shareapi_enforce_links_password_write_only" of app "core" has been set to "yes"
+    And user "user1" has logged in using the webUI
+    When the user creates a new public link for folder "simple-folder" using the webUI with
+      | role | Uploader |
     Then the user should see an error message on the public link share dialog saying "Passwords are enforced for link shares"
     And the public link should not have been generated
 
-  @skip @yetToImplement
   Scenario: user creates a public link with read-write permission without entering share password while enforce password on read only public share is enforced
-    Given parameter "shareapi_enforce_links_password_read_only" of app "core" has been set to "yes"
+    Given the setting "shareapi_enforce_links_password_read_only" of app "core" has been set to "yes"
+    And user "user1" has logged in using the webUI
     When the user creates a new public link for folder "simple-folder" using the webUI with
-      | permission | read-write |
-    And the public accesses the last created public link using the webUI
+      | role | Editor |
+    And the public uses the webUI to access the last public link created by user "user1"
     Then file "lorem.txt" should be listed on the webUI
 
-  @skip @yetToImplement
+  @yetToImplement
   Scenario: public should be able to access the shared file through public link
-    When the user creates a new public link for file 'lorem.txt' using the webUI
-    And the public accesses the last created public link using the webUI
-    Then the text preview of the public link should contain "Lorem ipsum dolor sit amet, consectetur"
-    And the content of the file shared by the last public link should be the same as "lorem.txt"
+    Given user "user1" has logged in using the webUI
+    And user "user1" has created a share with following settings
+      | path       | lorem.txt        |
+      | name       | Public link      |
+      | shareType  | 3                |
```

Comment:
If possible try to use language in shareType as ```shareType= public_link```. Check ```sharingHelper.js```

Reply:
:paintbrush: